pkg/plugins/policies/meshloadbalancingstrategy/plugin/v1alpha1/plugin.go

```diff
@@ -26,6 +35,91 @@ func (p plugin) MatchedPolicies(dataplane *core_mesh.DataplaneResource, resource
 }
 
 func (p plugin) Apply(rs *core_xds.ResourceSet, ctx xds_context.Context, proxy *core_xds.Proxy) error {
-	log.Info("apply is not implemented")
+	policies, ok := proxy.Policies.Dynamic[api.MeshLoadBalancingStrategyType]
+	if !ok {
+		return nil
+	}
+
+	listeners := policies_xds.GatherListeners(rs)
+	clusters := policies_xds.GatherClusters(rs)
+
+	for _, outbound := range proxy.Dataplane.Spec.Networking.GetOutbound() {
+		oface := proxy.Dataplane.Spec.Networking.ToOutboundInterface(outbound)
+		serviceName := outbound.GetTagsIncludingLegacy()[mesh_proto.ServiceTag]
+
+		computed := policies.ToRules.Rules.Compute(core_xds.MeshService(serviceName))
+		if computed == nil {
+			continue
+		}
+
+		conf := computed.Conf.(api.Conf)
+
+		if listener, ok := listeners.Outbound[oface]; ok {
```

Comment:
Should we create a set of listeners/clusters to avoid modifying more than once? Like in #5643

Reply:
Right, I split configuration in 2 loops so we don't configure clusters twice